### ISSUE TITLE
disk_resource_policies - not working properly

### ISSUE THREAD

**Issue by pawelrosada** [terraform-google-modules/terraform-google-vm#440]
### TL;DR

The issue occurs when attaching a snapshot resource policy to a disk in a VM instance template using Terraform. Although the configuration specifies the snapshot policy, the disks created from the template do not get the policy applied as expected.

### Expected behavior

When updating the VM instance template, the disks should automatically attach the specified snapshot resource policy, so that snapshots are managed according to the policy.

### Observed behavior

The disks are created but do not have the snapshot resource policy attached, even though it is specified in the configuration. When the template is copied, the disks are visible but lack the associated snapshot policy.

### Terraform Configuration

```hcl
module "vm_instance_template_db" {
  source     = "terraform-google-modules/vm/google//modules/instance_template"
  version    = "12.0.0"
  project_id = var.project
  region     = var.region

  disk_labels            = {}
  source_image_project   = "debian-cloud"
  source_image           = "debian-11"
  auto_delete            = false
  disk_type              = "pd-ssd"
  disk_size_gb           = 30
  network                = module.vpc.network_name
  subnetwork             = module.vpc.subnets_names[0]
  subnetwork_project     = var.project
  tags                   = ["db"]
  machine_type           = "e2-standard-4"
  name_prefix            = "db-${var.suffix}"
  disk_resource_policies = [google_compute_resource_policy.snapshot-policy.id]

  service_account = {
    email  = google_service_account.vm.email
    scopes = ["cloud-platform"]
  }

  access_config = [{
    nat_ip       = null
    network_tier = null
  }]

  depends_on = [module.vpc]
}

resource "google_compute_resource_policy" "snapshot-policy" {
  name    = "snapshot-policy"
  region  = var.region
  project = var.project
  snapshot_schedule_policy {
    schedule {
      hourly_schedule {
        hours_in_cycle = 4
        start_time     = "23:00"
      }
    }
    retention_policy {
      max_retention_days    = 7
      on_source_disk_delete = "KEEP_AUTO_SNAPSHOTS"
    }
    snapshot_properties {
      storage_locations = ["eu"]
      guest_flush       = true
    }
  }
}
```

### Terraform Version

```sh
v1.9.8
```
